File: WEB/pacman/script/model/Position.js
/**
 * 
 */
class Position {
    /**
     * 
     * @param {*} row the line of the position object
     * @param {*} column the column of the position object
     */
    constructor(row, column) {
        this._row = row;
        this._column = column;

    }

    /**
      * @returns {Integer} the unique row of the constructor
      */
    get row() { return this._row; }

    /**
      * @returns {Integer} the unique column of the constructor
      */
    get column() { return this._column; }

    /**
     * return a new position, add the position direction of row side and column side
     * @param {Direction} dir 
     * @returns returns the next position
     */
    nextPosition(dir) {
        return new Position(this._row + dir._deltaRow, this._column + dir._deltaColumn);
    }

}

File: JAVA/lucky-numbers/lucky-numbers-56359-master/src/main/java/g56359/luckynumbers/model/Position.java
package g56359.luckynumbers.model;

public class Position {

    private int row;
    private int column;

    /**
     *
     * @return the row of the position
     */
    public int getRow() {
        return row;
    }

    /**
     * @return the column of the position
     */
    public int getColumn() {
        return column;
    }

    /**
     *
     * @param row the row of the position
     * @param column the column of the position
     */
    public Position(int row, int column) {
        this.row = row;
        this.column = column;
    }
}
